snuba/query/allocation_policies/__init__.py

```diff
@@ -133,12 +180,30 @@ class AllocationPolicy(ABC, metaclass=RegisteredClass):
 
     def __init__(
         self,
-        storage_set_key: StorageSetKey,
+        storage_key: StorageKey,
         required_tenant_types: list[str],
         **kwargs: str,
     ) -> None:
         self._required_tenant_types = set(required_tenant_types)
-        self._storage_set_key = storage_set_key
```

Comment:
since you're removing the `storage_set_key`, please update the class docstring as well